src/ParseFile.js

```diff
@@ -418,39 +418,45 @@ const DefaultController = {
     if (source.format !== 'file') {
       throw new Error('saveFile can only be used with File-type sources.');
     }
-    // To directly upload a File, we use a REST-style AJAX request
-    const headers = {
-      'X-Parse-Application-ID': CoreManager.get('APPLICATION_ID'),
-      'Content-Type': source.type || (source.file ? source.file.type : null)
-    };
-    const jsKey = CoreManager.get('JAVASCRIPT_KEY');
-    if (jsKey) {
-      headers['X-Parse-JavaScript-Key'] = jsKey;
-    }
-    let sessionToken = options.sessionToken;
-    const userController = CoreManager.getUserController();
-    if (!sessionToken && userController) {
-      const currentUser = await userController.currentUserAsync();
-      sessionToken =  currentUser ? currentUser.getSessionToken() : undefined;
-    }
-    if (sessionToken) {
-      headers['X-Parse-Session-Token'] = sessionToken;
-    }
-    let url = CoreManager.get('SERVER_URL');
-    if (url[url.length - 1] !== '/') {
-      url += '/';
+    if (source.base64) {
+      // If base64 data has already been created, go ahead and save it.
+      source.format = 'base64';
+      return await DefaultController.saveBase64(name, source, options);
     }
-    url += 'files/' + name;
-    return CoreManager.getRESTController().ajax('POST', url, source.file, headers, options).then(res=>res.response)
+    const base64Data = await new Promise((res, rej) => {
+      // eslint-disable-next-line no-undef
+      const reader = new FileReader();
+      reader.readAsDataURL(source.file);
+      reader.onload = () => res(reader.result);
+      reader.onerror = error => rej(error);
+    });
```

Comment:
Would there be any overhead sending base64 to the server from client side? Just wanted to ask